dvc/repo/__init__.py

```diff
@@ -440,6 +440,8 @@ def is_dvc_internal(self, path):
     @contextmanager
     def open(self, path, remote=None, mode="r", encoding=None):
         """Opens a specified resource as a file descriptor"""
+        assert mode in {"r", "rt", "rb"}
```

Comment:
This assert guards incomplete implementation, this is why it was placed next to it. Moving it away will cause multiple issues:
- people might think that `SomeRemote.open(..., mode="w")` works
- when adding new remote it will be non-obvious how much functionality to `.open()` one must add
- extending implementation for one remote will lead to incoherence between guard and this new implementation.